extensions/amp-story-shopping/0.1/amp-story-shopping-config.js

```diff
@@ -1,19 +1,118 @@
 import {Services} from '#service';
 
+import {user} from '#utils/log';
+
 import {getElementConfig} from 'extensions/amp-story/1.0/request-utils';
 
+import {assertHttpsUrl} from '../../../src/url';
 import {
   Action,
   ShoppingConfigDataDef,
 } from '../../amp-story/1.0/amp-story-store-service';
 
+/**
+ * Max Length of strings allowed in shopping config.
+ */
+export const MAX_STR_LEN = 100;
+
+/** @const {!Object<string, !Array<function>>} */
+export const PRODUCT_VALIDATION_CONFIG = {
+  /* Required Attrs */
+  'productId': [validateRequired, validateStringLength],
+  'productTitle': [validateRequired, validateStringLength],
+  'productPrice': [validateRequired, validateNumber],
+  'productImages': [validateRequired, validateURLs],
+  'productPriceCurrency': [validateRequired, validateStringLength],
+  /* Optional Attrs */
+  'productColor': [validateStringLength],
+  'productSize': [validateStringLength],
+  'productIcon': [validateURLs],
+  'productTagText': [validateStringLength],
+  'reviewsData': [validateURLs],
+  'ctaText': [validateNumber],
+  'shippingText': [validateNumber],
+};
+
 /**
  * @typedef {{
  *  items: !Array<!ShoppingConfigDataDef>,
  * }}
  */
 let ShoppingConfigResponseDef;
 
+/**
+ * Validates if a required field exists for shopping config attributes
+ * @param {string} field
+ * @param {?string=} value
+ */
+export function validateRequired(field, value = null) {
+  if (value == null) {
+    throw Error(`Field ${field} is required.`);
+  }
+}
+
+/**
+ * Validates string length of shopping config attributes
+ * @param {string} field
+ * @param {?string=} str
+ */
+export function validateStringLength(field, str = null) {
+  if (str?.length > MAX_STR_LEN) {
```

Comment:
Why would all fields have a max length of 100 characters? For instance the price currency should be much shorter than that, right? And the product tag text wouldn't be able to fit 100 chars.

These limits could be more restrictive to ensure proper UX, if we want to enforce the correct length for these things, as having limits that are too permissive is not that useful.

Reply:
Doing characters limit is usually a bad idea and is unfair to different languages (e.g. it's totally different restrictions in Chinese vs German).
Instead, we should have a UI that correctly handles it (truncates text that's too long, or collapses some of it). I'd be against this max length limit as it sounds restrictive for no valid reason.

Reply:
After a discussion with @processprocess, I will go ahead and change the validation function to just check if it is a string and remove the string length check.